### ISSUE TITLE
Metrics

### ISSUE THREAD

**Issue by bouharb** [babelouest/ulfius#140]
Hello,

Is it possible to integrate a metrics tool like prometheus ?

Thanks in advance

**Comment by babelouest** [comment]
Hello @bouharb ,

That's a very good question. Unfortunately I don't know the tool so I would make a definitive assumption but I don't see why not.

I don't know exactly how this tool works but I see a [C++ library](https://github.com/jupp0r/prometheus-cpp). You can start with this one.

If you can integrate an Ulfius project with metrics, I'd be interested to get to know the result though.

**Comment by Citrullin** [comment]
Of course. Prometheus just needs an HTTP endpoint. So, define one and collect your metrics ;) 
Prometheus describes this also in the documentation: https://prometheus.io/docs/prometheus/latest/getting_started/
As I remember the structure is just ```key=value```

**Comment by babelouest** [comment]
Thanks @Citrullin , but the problem remains that I don't have the time these days to make a project like that...
Any help would be appreciated.

**Comment by babelouest** [comment]
Hello,

I'm digging out this issue from the past because I've recently used ulfius to create metrics endpoints for Prometheus in glewlwyd. The issue is available [here](https://github.com/babelouest/glewlwyd/issues/181)

Basically a new ulfius instance is created to host the metrics endpoint, and an asynchronous system with detached threads is used to increment the metrics.

**Comment by Citrullin** [comment]
> 
> 
> Thanks @Citrullin , but the problem remains that I don't have the time these days to make a project like that...
> Any help would be appreciated.

I don't think I was refering to you back then, but to the issue creator :) Afterall it's all open source :D

**Comment by babelouest** [comment]
My bad @Citrullin , sorry for the confusion :/
